fix(formatter): incorrect check for assignment-like node if the right-hand side is a member chain

Dunqing opened the pull request:
Previously, we didn't support MemberChain yet, so we missed this check. Now we have added.

Boshen commented:
### Merge activity

* **Jul 30, 3:09 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(formatter): incorrect check for assignment-like node if the right-hand side is a member chain (#12587)

Previously, we didn't support MemberChain yet, so we missed this check. Now we have added.
